### PR TITLE
DOC-4148: Update Data structure pages

### PR DESCRIPTION
**Pull request by mzorai**

### CODE REVIEW

File: docs/Platform_Development.rst [hunk 1]
```diff
@@ -1727,39 +1727,101 @@ See the following example:
...
+
+Social data is stored in the :ref:`JPA data source <Database.ConfiguringPLF>`
+in a set of database tables as the following:
+
+|image13|
+
+The table **SOC\_ACTIVITIES** stores all the activities and comments and has these attributes:
+
++-----------------------+-----------------+------------------------------------------------------------------------------------+
+| Property name         | Required type   | Description                                                                        |
```
Comment (thomasdelhomenie) [inline docs/Platform_Development.rst]:
```suggestion
| Name         | Type   | Description                                                                        |
```

File: docs/Platform_Development.rst [hunk 2]
```diff
@@ -1727,39 +1727,101 @@ See the following example:
...
+
+Social Data structure
+~~~~~~~~~~~~~~~~~~~~~~
+
+Social data is stored in the :ref:`JPA data source <Database.ConfiguringPLF>`
+in a set of database tables as the following:
+
+|image13|
+
+The table **SOC\_ACTIVITIES** stores all the activities and comments and has these attributes:
```
Comment (thomasdelhomenie) [inline docs/Platform_Development.rst]:
```suggestion
The table **SOC\_ACTIVITIES** stores all the activities and comments and has these fields:
```

File: docs/Platform_Development.rst [hunk 3]
```diff
@@ -1727,39 +1727,101 @@ See the following example:
...
+Data Structure
+================
 
 This section consists of the following main topics:
 
--  :ref:`Calendar JCR structure <PLFRefGuide.PLFDevelopment.JCRStructure.Calendar>`
+-  :ref:`Social Data structure <PLFRefGuide.PLFDevelopment.DataStructure.Social>`
+
+   A comprehensive knowledge of the Social Data Structure that is organized to 
+   conform to the data storage for the individual purpose of Social.
```
Comment (thomasdelhomenie) [inline docs/Platform_Development.rst]: I am not sure to understand the sentence. It should be simpler IMO, something like "A description of the Social Data Structure".

File: docs/Platform_Development.rst [hunk 4]
```diff
@@ -1727,39 +1727,101 @@ See the following example:
...
 
-.. _PLFRefGuide.PLFDevelopment.JCRStructure.Calendar:
+
+.. _PLFRefGuide.PLFDevelopment.DataStructure.Social:
+
+Social Data structure
+~~~~~~~~~~~~~~~~~~~~~~
+
+Social data is stored in the :ref:`JPA data source <Database.ConfiguringPLF>`
+in a set of database tables as the following:
```
Comment (thomasdelhomenie) [inline docs/Platform_Development.rst]: "as follows" ?

File: docs/Platform_Development.rst [hunk 5]
```diff
@@ -1727,39 +1727,101 @@ See the following example:
...
+
+|image13|
+
+The table **SOC\_ACTIVITIES** stores all the activities and comments and has these attributes:
+
++-----------------------+-----------------+------------------------------------------------------------------------------------+
+| Property name         | Required type   | Description                                                                        |
++=======================+=================+====================================================================================+
+| ACTIVITY_ID           | ID              | The unique ID of the activity.                                                     |
```
Comment (thomasdelhomenie) [inline docs/Platform_Development.rst]:
```suggestion
| ACTIVITY_ID           | ID              | The unique ID of the activity or comment.                                                     |
```

File: docs/Platform_Development.rst [hunk 6]
```diff
@@ -1786,35 +1848,35 @@ or the default ones in the system, it is stored under the **calendars**
...
-+-----------------------+-----------------+------------+------------------------------------------------------------------------------------------------------------------------------------------------------+
-| exo:privateUrl        | String          | false      | The private ICAL link of the calendar.                                                                                                               |
-+-----------------------+-----------------+------------+------------------------------------------------------------------------------------------------------------------------------------------------------+
++-----------------------+-----------------+------------------------------------------------------------------------------------------------------------------------------------------------------+
+| Field                 |          type   |  Description                                                                                                                                         |
```
Comment (thomasdelhomenie) [inline docs/Platform_Development.rst]: The calendar application is still in JCR, so we should not update this table IMO.

**Review by thomasdelhomenie — CHANGES_REQUESTED** [review]

File: docs/Platform_Development.rst (hunk 3 above)
Comment (mzorai) [reply docs/Platform_Development.rst]: OK

File: docs/Platform_Development.rst (hunk 6 above)
Comment (mzorai) [reply docs/Platform_Development.rst]: I did not change it, it looks like a bad copy paste of the table heading

Comment (thomasdelhomenie) [reply docs/Platform_Development.rst]: ok so you will revert it ?
It looks like a column has been removed also (Multiple).

Comment (mzorai) [reply docs/Platform_Development.rst]: The table structure was affected so it appears that the whole table was removed then added again (you notice this when you open whole file)

Comment (mzorai) [reply docs/Platform_Development.rst]: It should be OK now

File: docs/Platform_Development.rst [hunk 7]
```diff
@@ -2132,8 +2404,271 @@ stored in **CalendarApplication/calendarsetting**. The
...
+| DRAFT_PAGE_ID         | ID              | The unique ID of the wiki draft page attachment.                                   |
++-----------------------+-----------------+------------------------------------------------------------------------------------+
+| CREATED_DATE          | TIMESTAMP       | The creation date of the wiki draft page attachment.                               |
++-----------------------+-----------------+------------------------------------------------------------------------------------+
+| FULL_TITLE            | VARCHAR         | The title of the wiki draft page attachment.                                       |
++-----------------------+-----------------+------------------------------------------------------------------------------------+
+| ATTACHMENT_FILE_ID    | ID              | The unique ID of the file in the table FILES_FILES.                                |
++-----------------------+-----------------+------------------------------------------------------------------------------------+
+
+Table **WIKI\_WATCHERS*
```
Comment (thomasdelhomenie) [inline docs/Platform_Development.rst]:
```suggestion
Table **WIKI\_WATCHERS**
```

**Review by thomasdelhomenie — CHANGES_REQUESTED** [review]

File: docs/Platform_Development.rst (hunk 7 above)
Comment (mzorai) [reply docs/Platform_Development.rst]: (y) thank you!

**Comment by mzorai** [comment]
The content of this PR could be viewed here: https://docs-preprod.exoplatform.org/en/preproduction/Platform_Development.html#data-structure

**Comment by thomasdelhomenie** [comment]
Looks good. But actually 1 chapter is missing : the one related to the improvement Platform RDBMS :) It is only 1 table : LOGIN_HISTORY. I will give you the image of the table model.

**Comment by mzorai** [comment]
> Looks good. But actually 1 chapter is missing : the one related to the improvement Platform RDBMS :) It is only 1 table : LOGIN_HISTORY. I will give you the image of the table model.

OK i will add it thanks

**Comment by thomasdelhomenie** [comment]
Here it is : https://jira.exoplatform.org/secure/attachment/120458/login-history-data-model.png

File: docs/Platform_Development.rst [hunk 8]
```diff
@@ -3064,6 +3887,28 @@ type (exo:poll) has the following properties:
...
+The login history data is stored in this database table:
+
+|image19|
+
+Table **LOGIN\_HISTORY**
+
++-----------------------+-----------------+------------------------------------------------------------------------------------+
+| Name                  | Type            | Description                                                                        |
++=======================+=================+====================================================================================+
+| ID                    | ID              | The unique ID of the login history.                                                |
```
Comment (thomasdelhomenie) [inline docs/Platform_Development.rst]:
```suggestion
| ID                    | ID              | The unique ID of the login history entry.                                                |
```

File: docs/Platform_Development.rst [hunk 9]
```diff
@@ -3064,6 +3887,28 @@ type (exo:poll) has the following properties:
...
++-----------------------+-----------------+------------------------------------------------------------------------------------+
+| USER_ID               | VARCHAR         | The ID of related user to the login history.                                       |
```
Comment (thomasdelhomenie) [inline docs/Platform_Development.rst]:
```suggestion
| USER_ID               | VARCHAR         | The ID of related user to the login history entry.                                       |
```

File: docs/Platform_Development.rst [hunk 10]
```diff
@@ -3064,6 +3887,28 @@ type (exo:poll) has the following properties:
...
++-----------------------+-----------------+------------------------------------------------------------------------------------+
+| LOGIN_DATE            | TIMESTAMP       | The time of the last user's login date.                                            |
```
Comment (thomasdelhomenie) [inline docs/Platform_Development.rst]:
```suggestion
| LOGIN_DATE            | TIMESTAMP       | The date of the login history entry                                            |
```

**Review by thomasdelhomenie — CHANGES_REQUESTED** [review]

**Review by thomasdelhomenie — APPROVED** [review]

**Pull request merged**